sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,26 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
```

Comment:
> Their sum may still be smaller than the big eigenvalues, making it not a good regression test.

"May" is the term indeed. On the digits dataset this is not what is happening though. The reason we get infinite values and an error in master when calling `get_precision` is exactly because `np.all((self.explained_variance_ - self.noise_variance_) >= 0)` is `False` (because we took the sum of the eigenvalues rather than the mean).